octopusdeploy/schema_deployment_step.go

```diff
@@ -42,60 +44,41 @@ func expandDeploymentStep(flattenedStep map[string]interface{}) *deployments.Dep
 		step.Properties["Octopus.Action.MaxParallelism"] = core.NewPropertyValue(windowSize.(string), false)
 	}
 
-	if v, ok := flattenedStep["action"]; ok {
```

Comment:
There's a whole bunch of repetitive code here. Have pulled it out into local functions